src/main/java/com/revlearn/team1/controller/CourseController.java

```diff
@@ -0,0 +1,22 @@
+package com.revlearn.team1.controller;
+
+import com.revlearn.team1.model.Course;
+import com.revlearn.team1.service.CourseService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/Course")
+@RequiredArgsConstructor
+public class CourseController {
+    private final CourseService courseService;
+
+    @GetMapping("/{id}")
+    public Course getCourseById(@RequestParam Long courseId) {
+        return courseService.getById(courseId);
```

Comment:
can you A) wrap the return in 'ResponseEntity' to handle HTTP status codes
and
B) add a null check and return a 404 if course doesn't exist

Reply:
The HTTP status codes are handled by the exception handler in the Global Exception Handler.  Any errors will result in HTTP Response Entities in that file.

Reply:
oh ur right! Dope! ignore this comment then